src/pages/[platform]/build-a-backend/data/working-with-files/index.mdx

```diff
@@ -1447,29 +1405,19 @@ class SongViewModel: ObservableObject {
         }
 
         // Get the song record
-        let result = try await Amplify.API.mutate(request: .get(Song.self, byId: currentSong.id))
-        guard case .success(let queriedSong) = result else {
-            print("Failed with error: ", result)
-            return
-        }
-        guard let song = queriedSong else {
-            print("Song may have been deleted, no song by id: ", currentSong.id)
-            await setCurrentSong(nil)
+        guard var song = try await Amplify.API.mutate(request: .get(Song.self, byIdentifier: currentSong.id)).get() else {
+            print("Error: could not retrieve the song")
             return
         }
```

Comment:
this should be `Amplify.API.query` for better clarify

`Amplify.API.mutate` and `Amplify.API.query` technically has no differences, the "mutation" and "query" is specified in the GraphQLRequest document. Maybe this should have been one API like `Amplify.API.execute|run|perform`

Reply:
```suggestion
        guard var song = try await Amplify.API.mutate(request: .get(Song.self, byIdentifier: currentSong.id)).get() else {
            print("Song may have been deleted, no song by id: ", currentSong.id)
            return
        }
```